godot/src/decentraland_components/avatar_attach.gd

```diff
@@ -12,30 +15,68 @@ extends Node
 #  AAPT_RIGHT_HAND = 3;
 var attach_point: int = -1
 
+var _computed_colliders: Array[Node]
+var _main_scene_tree: SceneTree
 var _player_node: Node3D
 
 
-func _ready():
+func init():
+	var scene = SceneHelper.search_scene_node(self.get_parent())
+	scene.tree_changed.connect(self._on_tree_changed)
+
+	# Compute initial transform position
+	_process(0)
 	process_priority = 0
 
+	_main_scene_tree = get_tree()
 
-func _set_children_physics(node: Node, enable: bool):
-	for child in node.get_children():
-		_set_children_physics(child, enable)
+	var colliders: Array[Node] = []
+	get_collider_tree(self.get_parent(), colliders)
+	update_colliders(colliders)
 
+
+func get_collider_tree(node: Node, colliders: Array[Node]) -> void:
+	for child in node.get_children():
+		get_collider_tree(child, colliders)
 		if child.has_meta("dcl_col"):
-			if enable:
-				child.collision_layer = child.get_meta("dcl_col")
-			else:
-				child.collision_layer = 0
+			colliders.push_back(child)
 
 
-func _exit_tree():
-	_set_children_physics(self.get_parent(), true)
+func update_colliders(current_colliders: Array[Node]) -> void:
+	for collider in current_colliders:
+		var col_index := _computed_colliders.find(collider)
+		if col_index != -1:
+			_computed_colliders.remove_at(col_index)
+		else:
+			collider.collision_layer = 0
+
+	async_apply_after_physics(_computed_colliders.duplicate())
+
+	_computed_colliders = current_colliders
+
 
+func async_apply_after_physics(_computed_colliders: Array[Node]):
+	await _main_scene_tree.create_timer(TIME_TO_RETURN_BACK_LAYER).timeout
 
-func _enter_tree():
-	_set_children_physics(self.get_parent(), false)
+	for old_collider in _computed_colliders:
+		# Corner case when is moved from one avatar attach to other
+		if not is_parented_with_avatar_attach(old_collider):
+			old_collider.collision_layer = old_collider.get_meta("dcl_col")
+
+
+func is_parented_with_avatar_attach(_collider: Node) -> bool:
+	# TODO
+	return false
```

Comment:
```suggestion
func is_parented_with_avatar_attach(_collider: Node) -> bool:
	# TODO
	return false
```

what is the purpose of this func?

Reply:
it should return true when the collider is an nth-level child of some avatar attach.
the purpose of the usage above is to avoid removing collider (see comment)